### ISSUE TITLE
swiftlint:disable:file identifier_name not working via Terminal

### ISSUE THREAD

**Issue by DaniSpringer** [realm/SwiftLint#4865]
**(EDITED, please scroll to bottom of page)**

In Xcode, I have a file like:

```
... more code above ...
// swiftlint:disable:file identifier_name
/// An extension to provide conversion to and from CIELAB colors.
extension UIColor {

    /// The CIELAB components of a color - lightness (L) and chromaticity (a,b).
    struct CIELAB: Hashable {

        /// The lightness component of the color, in the range [0, 100] (darkest to brightest).
        var L: CGFloat
        /// The green-red chromaticity component of the color, typically in range [-128, 128].
        var a: CGFloat
        /// The blue-yellow chromaticity component of the color, typically in range [-128, 128].
        var b: CGFloat

    }
...
```
And while SwiftLint in Xcode doesn't show any warnings, if I run SwiftLint via Terminal:

```
swiftlint --reporter relative-path --quiet
```

Then it returns:

```
pickerfull/PickerFull/Model/UIColor+Extensions.swift:174:9: error: Identifier Name Violation: Variable name 'L' should be between 3 and 40 characters long (identifier_name)
```

It also returns a bunch of weird errors like:

```
objc[15634]: Class _TtC11SwiftSyntax16BumpPtrAllocator is implemented in both /opt/homebrew/Cellar/swiftlint/0.51.0/bin/swiftlint (0x100f33870) and /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/libSwiftSyntax.dylib (0x1049c4a10). One of the two will be used. Which one is undefined.
```

But maybe that should be a separate issue. Not sure.

Oddly, it also outputs this:

```
PickerFull/Model/UIColor+Extensions.swift:165:42: warning: Invalid SwiftLint Command Violation: swiftlint command does not have a valid action or modifier (invalid_swiftlint_command)
```

But once again, in Xcode, none of these are shown.

By the way, all code is here: https://github.com/DaniSpringer/pickerfull

**Comment by keith** [comment]
Are you sure you're running swiftlint in Xcode? Or at least the same version? The issue is `disable:file` isn't a thing, you want `// swiftlint:disable identifier_name` to do it for the whole file

**Comment by DaniSpringer** [comment]
Well, Xcode gives no warnings for:

```
// swiftlint:disable:file identifier_name
```

But if I try to change it to an invalid command by removing the final "e":

```
// swiftlint:disable:file identifier_nam
```

Then Xcode gives:

```
Superfluous Disable Command Violation: 'identifier_nam' is not a valid SwiftLint rule. Please remove it from the disable command. (superfluous_disable_command)
```

How do I check if SwiftLint is being used? And how can I check what version is being used in Xcode?

**Comment by keith** [comment]
I believe at some point swiftlint was incorrectly handling invalid identifiers like this, so it's possible it's just a version mismatch. In the shell script you're using to run swiftlint you can add `which swiftlint` and `swiftlint version` and view the output in the log. Or if you're using plugin you can view the version in your Package.swift and use those commands on the command line to see if your globally installed version is newer

**Comment by DaniSpringer** [comment]
If I run --version in Terminal I get the latest.
I actually tried looking up the Xcode version as mentioned here https://github.com/realm/SwiftLint/issues/933#issuecomment-265273959

But cannot get any output in Xcode console or Terminal.

I have a script in Xcode like this:

```
if which swiftlint >/dev/null; then
swiftlint
else
echo "warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint"
fi
```

And when running from Terminal I simply type "swiftlint", or, more specifically:

```
swiftlint --reporter relative-path --quiet
```

I tried adding to the Xcode script something like `swiftlint version` or `swiftlint --version` but got nothing in Terminal when running it again (nor in Xcode console).

**Comment by kenji21** [comment]
The :file modifier doesn't exists.... (as of 0.51.0)  => 
https://realm.github.io/SwiftLint/Structs/Command/Modifier.html#/s:13SwiftLintCore7CommandV8ModifierO8previousyA2EmF

**Comment by jpsim** [comment]
@kenji21 as Keith mentioned above, `disable:file` isn't a thing and never was, we just recently improved the validation of SwiftLint comment commands.

Please open a new issue if you're still encountering problems.
